### ISSUE TITLE
Add AppendInstance<TService> and Append<TService>(Func<TService> instanceCreator, Lifestyle lifestyte) to Collection

### ISSUE THREAD

**Issue by Metaphis** [simpleinjector/SimpleInjector#603]
Registering a single instance as part of a collection or a single singleton requires to supply a registration to to the `Append(Type, Registration)` method. Registrations are not well documented and figuring out how to accomplish this is difficult. These two overloads would provide the same set of registrations options to a collection as they would to the container itself.

Alternatively doing work to make creating registration more straightforward could work around this issue.

**Comment by dotnetjunkie** [comment]
Great suggestion.

In the meantime, you can achieve this by doing the following:

``` c#
container.Collection.Append(typeof(ISomeType),
    Lifestyle.Singleton.CreateRegistration<ISomeType>(() => new SomeTypeImpl(), container));
```

**Comment by Metaphis** [comment]
Thanks for the code sample. I've done exactly that. One note here is that the Container on the `ContainerCollectionRegistrator` is private. It would make for nicer extensions if the container was exposed so that the extension could be written this way:

```
public static void Append<T>(
    this ContainerCollectionRegistrator registrator, T instance)
    where T : class
{
    registrator.Append(typeof(T),
        Lifestyle.Singleton.CreateRegistration(() => instance, registrator.Container));
}
```

**Comment by dotnetjunkie** [comment]
>  It would make for nicer extensions if the container was exposed so that the extension could be written this way

I came to the same conclusion while writing my previous extension method.

**Comment by dotnetjunkie** [comment]
Feature branch [feature-603](https://github.com/simpleinjector/SimpleInjector/tree/feature-603) created.

**Comment by dotnetjunkie** [comment]
@Metaphis, I just released [4.5.0-beta1](https://github.com/simpleinjector/SimpleInjector/releases/tag/v4.5.0-beta1) that adds three new extension methods on `Container.Collection`.  Feel free to check this beta release out and give feedback on that feature (now you still can).
